apps/sertra/components/PackageDataFetcher.tsx

```diff
@@ -0,0 +1,65 @@
+import { useEffect, useState } from "react";
+import { ListingMod, TSListingMod } from "../api/models";
+import { ApiURLs, TsApiURLs } from "../api/urls";
+
+// TODO: To be replaced with ServerListingDetailData when the listing will contain an actual
+// object instead of an list of strings
+
+interface TempModRefType {
+  mods: string[];
+}
+
+export const FetchListingData = async (listingId: string) => {
+  const [listing, setListing] = useState<TempModRefType>();
+  useEffect(() => {
+    async function fetchListing() {
+      return await fetch(ApiURLs.ServerDetail(listingId));
+    }
+    async function jsonifyListing() {
+      setListing(await (await fetchListing()).json());
+    }
+    jsonifyListing();
+  });
```

Comment:
Missing the second parameter (to the `useEffect` function), which should be `[listingId]` in this case. Look up how it works in case you're unfamiliar with what it does.